pkg/registryclient/client.go

```diff
@@ -138,7 +138,7 @@ func WithCredentialProviders(credentialProviders ...string) Option {
 		var chains []authn.Keychain
 		helpers := sets.New(credentialProviders...)
 		if helpers.Has("default") {
-			chains = append(chains, authn.DefaultKeychain)
+			chains = append(chains, RegistryKeychain)
```

Comment:
This should be keychain and not registry keychain, we dont want to expose the attributes in Registry keychain